src/explorer/CosmosDBTreeItem.ts

```diff
@@ -84,31 +86,31 @@ export class CosmosDBTreeItem extends AzureParentTreeItem<ISiteTreeRoot> {
 
     public async createChildImpl(showCreatingTreeItem: (label: string) => void): Promise<AzureTreeItem<ISiteTreeRoot>> {
         const databaseToAdd = await ext.cosmosAPI.pickTreeItem({
-            resourceType: 'Database',
-            apiType: ['Mongo']
+            resourceType: 'Database'
         });
         if (!databaseToAdd) {
             throw new UserCancelledError();
         }
         const appSettings = await this.root.client.listApplicationSettings();
-        const appSettingKeyToAdd: string = await ext.ui.showInputBox({
-            prompt: 'Enter new connection setting key',
-            validateInput: (v?: string): string | undefined => validateAppSettingKey(appSettings, v),
-            value: "MONGO_URL"
-        });
+        const newAppSettings: Map<string, string> = databaseToAdd.docDBData ? await this.promptForDocDBAppSettings(appSettings, databaseToAdd) : await this.promptForMongoAppSettings(appSettings, databaseToAdd);
+
         // tslint:disable-next-line:strict-boolean-expressions
         appSettings.properties = appSettings.properties || {};
-        appSettings.properties[appSettingKeyToAdd] = databaseToAdd.connectionString;
+        for (const [k, v] of newAppSettings) {
+            appSettings.properties[k] = v;
+        }
+
         await this.root.client.updateApplicationSettings(appSettings);
         await this.parent.parent.appSettingsNode.refresh();
 
-        const createdDatabase = new CosmosDBConnection(this, databaseToAdd, appSettingKeyToAdd);
+        const createdDatabase = new CosmosDBConnection(this, databaseToAdd, newAppSettings.keys()[0]);
```

Comment:
Clarifying comment as to what is going on here?

Reply:
I assume you're just confused by why I'm doing `[0]`?

Refactored how this works. Let me know if it's still confusing.